examples/intRVFL.py

```diff
@@ -0,0 +1,539 @@
+                # DK: Point for extension. Currently, it assumes implicitly that there is only one vector (prototype) per class, which does not have to be the case in general
+                predictions = torch.argmax(outputs, dim=-1)
+                accuracy.update(predictions.cpu(), labels)
+
+        return accuracy.compute().item()
+
+    def evaluate(self):
+        # For all datasets in the collection
+        for i in range(len(self.dataset_collection)):
+            # Fetch the current dataset
+            dataset = getattr(torchhd.datasets, self.dataset_collection[i])
+
+            # Run for the requested number of simulations
+            for sim in range(self.simul):
+                # If no separate test dataset available - do 4-fold cross-validation
+                if hasattr(dataset, "num_folds"):
+                    accuracy_dataset = 0.0
+                    for fold_id in range(dataset.num_folds):
+                        # Set test and train datasets for the current fold
+                        train_ds = dataset(
+                            "../data", train=True, download=True, fold=fold_id
+                        )
+                        test_ds = dataset(
+                            "../data", train=False, download=False, fold=fold_id
+                        )
+
+                        # Initialize model
+                        model = self.model_class(
+                            dataset, train_ds[0][0].size(-1), self.device
+                        ).to(self.device)
+                        # Train model and computer the validation accuracy
+                        accuracy_dataset += self.compute_accuracy(
+                            model, train_ds, test_ds
+                        )
+
+                    # Average over folds
+                    accuracy_dataset = accuracy_dataset / dataset.num_folds
+
+                # Case of avaiable test set
+                else:
+                    # Set test and train datasets
+                    train_ds = dataset("../data", train=True, download=True)
+                    test_ds = dataset("../data", train=False, download=False)
+
+                    # Initialize model
+                    model = self.model_class(
+                        dataset, train_ds[0][0].size(-1), self.device
+                    ).to(self.device)
+                    # Train model and computer the test accuracy
+                    accuracy_dataset = self.compute_accuracy(model, train_ds, test_ds)
+
+                print(
+                    f"Dataset {i} - {self.dataset_collection[i]}: average testing accuracy of {(accuracy_dataset* 100):.3f}%"
+                )
+
+                # Update the statistics for the current dataset and simulation
+                self.accuracy_collection[i, sim] = accuracy_dataset
+
+        # DK: Think whether to average accross simulations or not. May be introduce boolean variable indicating whether to do so or not
```

Comment:
I would not do it, it's a one-liner to average but if someone wants the standard deviation it's impossible from the average. And more options is more things to maintain.

Reply:
Removed the comment